runtime.c

```diff
@@ -153,38 +153,36 @@ bool check_nt(bam1_t *read, int32_t position, unsigned char nt, bool exact)
 	if (read->core.flag & BAM_FUNMAP) {
 		return false;
 	}
-	if (read->core.pos > position || compute_mapped_end(read) < position) {
+	if (read->core.pos +1 > position || compute_mapped_end(read) < position) {
 		return false;
 	}
-	if ((read->core.flag & BAM_FUNMAP) || read->core.n_cigar == 0) {
-		mapped_position =
-		    (read->core.flag & BAM_FREVERSE) ? (read->core.l_qseq -
-							position +
-							read->core.pos -
-							1) : (position -
-							      read->core.pos);
+
+	if ( read->core.n_cigar == 0) {
+			mapped_position = bam_is_rev(read) ? (read->core.l_qseq - position + read->core.pos) : (position - read->core.pos-1);
```

Comment:
I found a bug in this while I was writing a test case yesterday. I had to find the end of the sequence using the computed cigar rather than by the actual sequence length. It is updated in the latest commit to the branch.